addons/mass_mailing/models/mass_mailing.py

```diff
@@ -746,34 +583,34 @@ def get_recipients(self, cr, uid, mailing, context=None):
             res_ids = random.sample(remaining, topick)
         return res_ids
 
-    def get_remaining_recipients(self, cr, uid, mailing, context=None):
-        res_ids = self.get_recipients(cr, uid, mailing, context=context)
-        already_mailed = self.pool['mail.mail.statistics'].search_read(cr, uid, [('model', '=', mailing.mailing_model),
-                                                                                 ('res_id', 'in', res_ids),
-                                                                                 ('mass_mailing_id', '=', mailing.id)], ['res_id'], context=context)
+    def get_remaining_recipients(self):
+        res_ids = self.get_recipients()
+        already_mailed = self.env['mail.mail.statistics'].search_read([('model', '=', self.mailing_model),
+                                                                     ('res_id', 'in', res_ids),
+                                                                     ('mass_mailing_id', '=', self.id)], ['res_id'])
         already_mailed_res_ids = [record['res_id'] for record in already_mailed]
         return list(set(res_ids) - set(already_mailed_res_ids))
 
-    def send_mail(self, cr, uid, ids, context=None):
-        author_id = self.pool['res.users'].browse(cr, uid, uid, context=context).partner_id.id
-        for mailing in self.browse(cr, uid, ids, context=context):
+    def send_mail(self):
+        author_id = self.env.user.partner_id.id
+        for mailing in self:
             # instantiate an email composer + send emails
-            res_ids = self.get_remaining_recipients(cr, uid, mailing, context=context)
+            res_ids = mailing.get_remaining_recipients()
             if not res_ids:
                 raise UserError(_('Please select recipients.'))
 
-            if context:
-                comp_ctx = dict(context, active_ids=res_ids)
+            if self.env.context:
+                comp_ctx = dict(self.env.context, active_ids=res_ids)
             else:
```

Comment:
Now context will never be None. We can remove this and put `active_ids=res_ids` in the `with_context`